libs/utils/src/pq_proto.rs

```diff
@@ -788,18 +677,17 @@ impl<'a> BeMessage<'a> {
                 buf.put_u8(b'E');
                 write_body(buf, |buf| {
                     buf.put_u8(b'S'); // severity
-                    write_cstr(&Bytes::from("ERROR"), buf)?;
+                    buf.put_slice(b"ERROR\0");
```

Comment:
It's easy to forget about the null terminator. I think `write_cstr` would still be a good idea here. (or a variant of `write_cstr` that can take a plain &[u8] as argument instead of Bytes)

Reply:
I get your point, but I'll stick with my changes.

Your logic implies that for some reason it's hard(er) to forget to use `write_cstr`, which is a sentiment I can't agree with. The real problem is that, excluding tests, in these circumstances there's no entity that could really enforce this invariant. No syntax sugar can make up for this.

Reply:
Ok